docs/deploying.adoc

```diff
@@ -9,11 +9,250 @@ Put the filter jars in some directory
+used to decrypt the key too.
+<4> (Optional) Store type. Supported types are: `PKCS12`, `JKS` and `PEM`.  Defaults to Java default key store type
+which is usually `PKCS12`.
+
+Alternatively, if your key material is in separate PEM files (private key, and certificate/intermediates), the following
+configuration may be used:
+
+[source, yaml]
+----
+virtualClusters:
+  demo:
+    tls:
+        key:
+          privateKeyFile: /opt/cert/server.key          # <1>
+          certificateFile: /opt/cert/server.crt         # <2>
+          keyPassword:
+            passwordFile: /opt/cert/key.password        # <3>
+    clusterNetworkAddressConfigProvider:
+      ...
+----
+<1> File system location of the server private key.
+<2> File system location of the server certificate and intermediate(s).
+<3> (Optional) File system location of a file containing the key's password.
+
+=== Upstream TLS
+
+Here's how to enable TLS for the upstream side.  This means that Kroxylicious connects to the (physical) Kafka Cluster)
+over TLS.  For this, your Kafka Cluster must have already been configured to use TLS.
+
+By default, Kroxylicious uses the trust of the platform to decide whether the Kafka Cluster is trusted or not.  To
+support cases where trust must be overridden (such as use-cases involving the use of private CAs or self-signed
+certificates), Kroxylicious accepts override trust material in PKCS12 or JKS keystore format, or PEM formatted certificates.
+
+The following illustrates enabling TKS using platform trust:
+
+[source, yaml]
+----
+virtualClusters:
+  demo:
+    targetCluster:
+      bootstrap_servers: myprivatecluster:9092
+      tls: {}                                         <1>
+      #...
+----
+<1> Use an empty object to enable TLS using platform trust.
+
+In case where the trust needs to be overridden, use configuration like this:
+
+[source, yaml]
+----
+virtualClusters:
+  demo:
+    targetCluster:
+      bootstrap_servers: myprivatecluster:9092
+      tls:
+        trust:
+          storeFile: /opt/cert/trust.p12                # <1>
+          storePassword:
+            passwordFile: /opt/cert/store.password      # <2>
+          storeType: PKCS12                             # <3>
+      #...
+----
+<1> File system location of a truststore (or in the case of `PEM` format a text file containing the certificates).
+<2> File system location of a file containing the trust store's password.
+<3> (Optional) Trust store type. Supported types are: `PKCS12`, `JKS` and `PEM`.  Defaults to Java default key store type (PKCS12).
+
+It is also possible to disable trust so that Kroxylicious will connect to any Kafka Cluster regardless of its certificate
+validity.
+
+WARNING: This option is not recommended for production use.
+
+[source, yaml]
+----
+virtualClusters:
+  demo:
+    targetCluster:
+      bootstrap_servers: myprivatecluster:9092
+      tls:
+        trust:
+          insecure: true                                # <1>
```

Comment:
Have been meaning to mention this for a while. 

```yaml
trust:
          insecure: true
```

Doesn't make sense to me as a config language. Maybe this?

```yaml
trust:
          all: true
```

Reply:
Hmmm. Not sure. `wget` is the best IMHO but insecure is common enough. So lets leave it.